### DISCUSSION TITLE
Team D - Challengers 3주차

### DISCUSSION THREAD

**Discussion by 1hk-star** [lablup/contribution-academy-2021#68] [Show and tell]
.

**Comment by 1hk-star** [comment]
1. 코스 3 수행
2. backend.ai-webui 설치하기
![image](https://user-images.githubusercontent.com/32287560/130328353-41ea8e1b-0d17-4f42-9bf0-4ec05158d656.png)
3. 나만의 문장으로 각 컴포넌트 설명해보기
Web Server - 사용자 웹브라우저 또는 Client SDK가 네트워크 스토리지로부터의 대용량 입출력을 바로 할 수 있도록 해주는 서비스
Storage Proxy - Web UI와 SPA (single-page app) 구현을 위한 라우팅을 제공하고 웹 세션 기반 사용자 인증을 제공하는 HTTP 서비스

**Comment by youngjun0627** [comment]
1. [코스 3 진행완료](https://github.com/lablup/contribution-academy-2021/discussions/54#discussioncomment-1188688)
2. webui install
![image](https://user-images.githubusercontent.com/57058726/130339476-ea1c7602-bd02-46ca-87de-eddf57c6a16a.png)
---
webui endpoint 설정 후 login까지 한 화면
![image](https://user-images.githubusercontent.com/57058726/130573081-cd68a9a9-641e-4325-9eba-8f8d2b0f759d.png)

3. 나만의 문장으로 각 컴포넌트 설명해보기
- webui : 타입스크립트 및 자바스크립트를 이용하여 App 및 web service에서 사용자에게 편리한 기능과 그래픽을 제공한다.
- kernels : 다양한 언어 및 프레임워크를 최적화된 환경으로 제공해주는 docker images의 보관소이다

**Comment by hyeyeon-sun** [comment]
1. 코스 3 agent - aiohttp error 해결중입니다.

2. web ui - electron install이 제대로 되지 않아 메인서버만 돌려보았습니다. 

![image](https://user-images.githubusercontent.com/39080868/130349531-eb8db4b8-5913-45da-8c13-99c74464e657.png)
<br>
3. 나만의 문장으로 각 컴포넌트 설명해보기
 - agent : 하드웨어 위에 깔려서 gpu 가상화 및 보안 레이어를 제공한다. 상태 모니터, watcher, 통계를 모으는 Statistics collector을 가지고 있다.
 - Client SDK for Python : 유저가 웹서버와 app proxy, storage proxy에 접근할 수 있게 해주며, 이 SDK를 기반으로 만들어진 CLI와 GUI를 통해  backend.ai를 사용할 수 있다.

**Comment by JHyunB** [comment]
1. 코스3 수행
2. 
![image](https://user-images.githubusercontent.com/36959292/130355511-591627b1-2c5b-4208-b716-7fd8ed713314.png)

3.
 - common : manager, agent, storage proxy 등 서버 측 컴포넌트들을 위한 공통 라이브러리
 - manager : agent들의 자원 모니터링 및 사용자 인증과 세션 실행등의 API를 제공
